src/Ext/HouseType/Hooks.cpp

```diff
@@ -0,0 +1,21 @@
+#include "Body.h"
+
+#include <InfantryTypeClass.h>
+#include <HouseClass.h>
+
+DEFINE_HOOK(0x4430BC, BuildingClass_RegisterDestruction_EjectCountryCrew, 0x6)
+{
+	GET(InfantryTypeClass*, pInfType, EAX);
+
+	GET(BuildingClass*, pThis, EDI);
+	HouseTypeClass* pHouseType = pThis->Owner->Type;
+	auto pExt = HouseTypeExt::ExtMap.Find(pHouseType);
+
+	if (pExt->CountryCrew)
+	{
+		pInfType = pExt->CountryCrew_Type;
+		R->EAX(pInfType);
+	}
```

Comment:
```suggestion
	GET(InfantryTypeClass*, pInfType, EAX);
	GET(BuildingClass*, pThis, EDI);
	auto pExt = HouseTypeExt::ExtMap.Find(pThis->Owner->Type);

	if (pExt->CountryCrew)
		R->EAX(pExt->CountryCrew_Type.Get(pInfType));
```
Well, after the suggestion, `pExt->CountryCrew` can be deleted totally, it is useless. But I leave it here now, the implementation maybe need more consideration.